--- netcdf-4.1.3/nf_test/f90tst_vars2.f90.orig	2011-06-10 15:35:01.000000000 +0200
+++ netcdf-4.1.3/nf_test/f90tst_vars2.f90	2011-06-25 09:04:21.118289857 +0200
@@ -26,7 +26,6 @@
   integer :: x_dimid, y_dimid
   integer :: nvars, ngatts, ndims, unlimdimid, file_format
   integer :: x, y
-  integer, parameter :: CACHE_NELEMS = 10000, CACHE_SIZE = 1000000
   integer, parameter :: DEFLATE_LEVEL = 4
   integer (kind = 8), parameter :: TOE_SAN_VALUE = 2147483648_8
   character (len = *), parameter :: VAR1_NAME = "Chon-Ji"
@@ -74,8 +73,7 @@
   call check(nf90_def_var(ncid, VAR2_NAME, NF90_INT, dimids, varid2, contiguous = .TRUE.))
   call check(nf90_def_var(ncid, VAR3_NAME, NF90_INT64, varid3))
   call check(nf90_def_var(ncid, VAR4_NAME, NF90_INT, x_dimid, varid4, contiguous = .TRUE.))
-  call check(nf90_def_var(ncid, VAR5_NAME, NF90_INT, dimids, varid5, chunksizes = chunksizes, &
-       option_mask = nf90_szip_ec_option_mask, bits_per_pixel = 16))
+  call check(nf90_def_var(ncid, VAR5_NAME, NF90_INT, dimids, varid5, chunksizes = chunksizes))
 
   ! As a test, set and the check the chunk cache for one of these
   ! variables. Really there is no need for this, since the 1 MB cache
@@ -83,7 +82,7 @@
   ! per-var chunk cache really helps performance.
   call check(nf90_set_var_chunk_cache(ncid, varid1, CACHE_SIZE, &
        CACHE_NELEMS, CACHE_PREEMPTION))
-  call check(nf90_inquire_variable(ncid, varid1_in, cache_size = cache_size_in, &
+  call check(nf90_inquire_variable(ncid, varid1, cache_size = cache_size_in, &
        cache_nelems = cache_nelems_in, cache_preemption = cache_preemption_in))
   if (cache_size_in .ne. CACHE_SIZE .or. cache_nelems_in .ne. &
        CACHE_NELEMS .or. cache_preemption_in .ne. CACHE_PREEMPTION) stop 1
--- netcdf-4.1.3/f90/netcdf_visibility.f90.orig	2011-06-10 15:35:03.000000000 +0200
+++ netcdf-4.1.3/f90/netcdf_visibility.f90	2011-06-25 15:30:24.575732121 +0200
@@ -20,6 +20,6 @@
   public :: nf90_put_att, nf90_get_att
   
   ! Variable routines
-  public :: nf90_def_var, nf90_inq_varid, nf90_rename_var, nf90_inquire_variable 
+  public :: nf90_def_var, nf90_inq_varid, nf90_set_var_chunk_cache, nf90_rename_var, nf90_inquire_variable 
   ! overloaded functions
   public :: nf90_put_var, nf90_get_var
