--- netcdf-3.6.1/src/Makefile.orig	2006-01-30 19:56:02.000000000 +0100
+++ netcdf-3.6.1/src/Makefile	2006-02-05 13:16:02.551287500 +0100
@@ -30,11 +30,12 @@
 
 # I moved fortran to the bottom of the list, sometimes AIX make get's confused 
 # if the C library and the fortran library are built in the same second. - Ed
-all:		libsrc/all	\
+# but fortran objects are linked into C library, so must it be built first --q
+all:		fortran/all	\
+		libsrc/all	\
 		ncgen/all	\
 		ncdump/all	\
 		cxx/all \
-		fortran/all	\
 		f90/all	
 
 test:		fortran/all \
--- netcdf-3.6.1/src/cxx/Makefile.orig	2006-01-26 21:13:44.000000000 +0100
+++ netcdf-3.6.1/src/cxx/Makefile	2006-02-03 20:20:18.730874250 +0100
@@ -11,19 +11,19 @@
 		  nctst.cpp ncvalues.cpp ncvalues.h netcdf.cpp netcdfcpp.h \
 		  netcdf.hh install-sh ref_nctst.cdl ref_nctst_64bit_offset.cdl
 
-LIBRARY		= libnetcdf_c++.a
-lib_netcdf	= ../libsrc/libnetcdf.a
-ld_netcdf	= -L../libsrc -lnetcdf
+LIBRARYL	= libnetcdf_c++.la
+lib_netcdf	= ../libsrc/libnetcdf.la
+ld_netcdf	= -L../libsrc/.libs -lnetcdf
 
 HEADER1		= netcdfcpp.h
 HEADER2		= ncvalues.h
 HEADER3		= netcdf.hh
 
-libs		=  $(LIBRARY) $(lib_netcdf) $(LIBS)
+libs		=  -L.libs -lnetcdf_c++ $(ld_netcdf) $(LIBS)
 
 purify		= purify
 
-LIB_OBJS	= netcdf.o ncvalues.o
+LIB_OBJS	= netcdf.lo ncvalues.lo
 
 # a test program that uses some of the draft C++ interface
 prog		= nctst
@@ -33,8 +33,11 @@
 		  *.cps *.dvi *.fns \
 		  *.log *~ *.gs *.aux *.cp *.fn *.ky *.pg *.toc *.tp *.vr
 
-all:		$(LIBRARY) $(prog)
+all:		$(LIBRARYL) $(prog)
 
+libnetcdf_c++.la: $(LIB_OBJS)
+	libtool --mode=link --tag=CXX $(CXX) -o libnetcdf_c++.la $(LIB_OBJS) $(LDFLAGS) -rpath $(LIBDIR) $(lib_netcdf)
+  
 test:	 $(prog) FORCE
 	./$(prog)
 	../ncdump/ncdump -n ref_nctst nctst_classic.nc > nctst_classic.cdl
@@ -45,9 +48,9 @@
 
 check: test
 
-install:	$(LIBDIR)/$(LIBRARY) $(INCDIR)/$(HEADER1) $(INCDIR)/$(HEADER2)\
+install:	$(LIBRARYL) $(INCDIR)/$(HEADER1) $(INCDIR)/$(HEADER2)\
 		$(INCDIR)/$(HEADER3)
-
+	libtool --mode=install install $(LIBRARYL) $(LIBDIR)
 
 uninstall:
 	-rm -f $(LIBDIR)/$(LIBRARY)
@@ -55,7 +58,7 @@
 	-rm -f $(INCDIR)/$(HEADER2)
 	-rm -f $(INCDIR)/$(HEADER3)
 
-$(prog) : $(prog_objs) $(LIBRARY) $(lib_netcdf)
+$(prog) : $(prog_objs) $(LIBRARYL) $(lib_netcdf)
 	$(CXX) -o $@ $(CXXFLAGS) $(LDFLAGS) $(prog_objs) $(libs)
 
 nctst_src: nctst.cpp netcdf.cpp ncvalues.cpp
diff -Nur netcdf-3.5.0.orig/src/fortran/Makefile netcdf-3.5.0/src/fortran/Makefile
--- netcdf-3.5.0.orig/src/fortran/Makefile	Wed Feb 21 22:20:26 2001
+++ netcdf-3.5.0/src/fortran/Makefile	Wed Apr  4 23:18:04 2001
@@ -13,20 +13,20 @@
 ld_netcdf	= $(LIBRARY)
 HEADER		= netcdf.inc
 LIB_OBJS	= \
-		  fort-attio.o	\
-		  fort-control.o	\
-		  fort-dim.o	\
-		  fort-genatt.o	\
-		  fort-geninq.o	\
-		  fort-genvar.o	\
-		  fort-lib.o	\
-		  fort-misc.o	\
-		  fort-v2compat.o	\
-		  fort-vario.o	\
-		  fort-var1io.o	\
-		  fort-varaio.o	\
-		  fort-varmio.o	\
-		  fort-varsio.o
+		  fort-attio.lo	\
+		  fort-control.lo	\
+		  fort-dim.lo	\
+		  fort-genatt.lo	\
+		  fort-geninq.lo	\
+		  fort-genvar.lo	\
+		  fort-lib.lo	\
+		  fort-misc.lo	\
+		  fort-v2compat.lo	\
+		  fort-vario.lo	\
+		  fort-var1io.lo	\
+		  fort-varaio.lo	\
+		  fort-varmio.lo	\
+		  fort-varsio.lo
 #
 
 MANUAL		= netcdf.3f
@@ -69,7 +69,7 @@
 #
 
 
-all:		$(LIBRARY) $(MANUAL)
+all:		$(LIB_OBJS) $(MANUAL)
 
 test:		ftest FORCE
 	./ftest
--- netcdf-3.6.1/src/libsrc/Makefile.orig	2005-02-17 17:15:11.000000000 +0100
+++ netcdf-3.6.1/src/libsrc/Makefile	2006-02-03 20:50:28.179957750 +0100
@@ -7,7 +7,7 @@
 
 INCLUDES = -I.. -I.
 
-LIBRARY 	= libnetcdf.a
+LIBRARYL 	= libnetcdf.la
 ld_netcdf	= -L. -lnetcdf
 
 HEADER	= netcdf.h
@@ -53,7 +53,23 @@
 	t_ncxx.c \
 	t_ncx.c install-sh
 
-LIB_OBJS = $(LIB_CSRCS:.c=.o)
+LIB_OBJS = $(LIB_CSRCS:.c=.lo)
+
+LIB_FOBJS	= \
+		  ../fortran/fort-attio.lo	\
+		  ../fortran/fort-control.lo	\
+		  ../fortran/fort-dim.lo	\
+		  ../fortran/fort-genatt.lo	\
+		  ../fortran/fort-geninq.lo	\
+		  ../fortran/fort-genvar.lo	\
+		  ../fortran/fort-lib.lo	\
+		  ../fortran/fort-misc.lo	\
+		  ../fortran/fort-v2compat.lo	\
+		  ../fortran/fort-vario.lo	\
+		  ../fortran/fort-var1io.lo	\
+		  ../fortran/fort-varaio.lo	\
+		  ../fortran/fort-varmio.lo	\
+		  ../fortran/fort-varsio.lo
 
 GARBAGE		= t_ncio.o t_ncio t_ncx.o t_ncx t_ncxx.o t_ncxx \
 	t_nc.o t_nc test.nc *.so
@@ -61,15 +77,19 @@
 DIST_GARBAGE	= ncconfig.h
 
 
-all:		$(LIBRARY) $(MANUAL)
+all:		$(LIBRARYL) $(MANUAL)
 
 check:		full_test 
 
 full_test:	test_ncx test nctest
 
-install:	$(LIBDIR)/$(LIBRARY)	\
+install:	$(LIBRARYL)	\
 		$(INCDIR)/$(HEADER)	\
 		$(MANDIR)/man3/$(MANUAL)
+	libtool install $(LIBRARYL) $(LIBDIR)
+
+libnetcdf.la:	$(LIB_OBJS)
+	libtool --mode=link --tag=CC $(CC) -o libnetcdf.la $(LIB_OBJS) $(LIB_FOBJS) $(LDFLAGS) -rpath $(LIBDIR)
 
 uninstall:
 	-rm -f $(LIBDIR)/$(LIBRARY)
@@ -94,10 +114,10 @@
 	cmp test.nc test_nc.sav
 	@echo '*** Success ***'
 
-nctest:		$(LIBRARY)
+nctest:		$(LIBRARYL)
 	(cd ../nctest ; make test)
 
-nc_test:	$(LIBRARY)
+nc_test:	$(LIBRARYL)
 	(cd ../nc_test ; make test)
 
 
diff -Nur netcdf-3.5.0.orig/src/ncdump/Makefile netcdf-3.5.0/src/ncdump/Makefile
--- netcdf-3.5.0.orig/src/ncdump/Makefile	Tue Mar 13 18:42:52 2001
+++ netcdf-3.5.0/src/ncdump/Makefile	Wed Apr  4 23:22:41 2001
@@ -16,11 +16,11 @@
 		  test0.cdl ncdump.1 Makefile
 MANUAL		= ncdump.1
 
-lib_netcdf	= ../libsrc/libnetcdf.a
+lib_netcdf	= ../libsrc/libnetcdf.la
 
 # Don't use "-L../libsrc -lnetcdf" in the following because that doesn't
 # work on a CRAY T90 (sigh).
-ld_netcdf	= $(lib_netcdf)
+ld_netcdf	= -L../libsrc/.libs -lnetcdf
 
 OBJS		=  ncdump.o vardata.o dumplib.o
 GARBAGE		= $(PROGRAM) test0.nc test1.nc test1.cdl test2.cdl
diff -Nur netcdf-3.5.0.orig/src/ncgen/Makefile netcdf-3.5.0/src/ncgen/Makefile
--- netcdf-3.5.0.orig/src/ncgen/Makefile	Tue Mar 13 18:42:53 2001
+++ netcdf-3.5.0/src/ncgen/Makefile	Wed Apr  4 23:23:09 2001
@@ -15,10 +15,10 @@
 		  ctest.c ctest ctest0.nc ctest1.cdl \
 		  ftest.f ftest ftest0.nc ftest1.cdl
 
-lib_netcdf	= ../libsrc/libnetcdf.a
+lib_netcdf	= ../libsrc/libnetcdf.la
 # Don't use "-L../libsrc -lnetcdf" in the following because that doesn't
 # work on a CRAY T90 (sigh).
-ld_netcdf	= $(lib_netcdf)
+ld_netcdf	= -L../libsrc/.libs -lnetcdf
 
 PACKING_LIST	= Makefile depend escapes.c generic.h \
 		  ncgenyy.c ncgentab.c ncgentab.h \
diff -Nur netcdf-3.5.0.orig/src/rules.make netcdf-3.5.0/src/rules.make
--- netcdf-3.5.0.orig/src/rules.make	Mon Mar 12 23:31:55 2001
+++ netcdf-3.5.0/src/rules.make	Wed Apr  4 23:24:06 2001
@@ -4,7 +4,7 @@
 # It should be processed by every execution of the that utility.
 
 .SUFFIXES:
-.SUFFIXES:	.a .o .i .f .c .cpp .F .y .l .m4
+.SUFFIXES:	.a .o .lo .i .f .c .cpp .F .y .l .m4
 
 
 ################################################################################
@@ -13,11 +13,17 @@
 .c.o:
 	$(COMPILE.c) $<
 
+.c.lo:
+	libtool --mode=compile --tag=CC $(COMPILE.c) $<
+
 .c.i:
 	$(CPP) $(CPPFLAGS) $< >$@
 
 .cpp.o:
 	$(COMPILE.cxx) $<
+
+.cpp.lo:
+	libtool --mode=compile --tag=CXX $(COMPILE.cxx) $<
 
 # Not all FORTRAN compilers support C-preprocessing of *.F files; ergo, a 
 # relatively complicated rule ensues.
--- netcdf-3.6.1/src/f90/Makefile.orig	2004-12-18 19:08:01.000000000 +0100
+++ netcdf-3.6.1/src/f90/Makefile	2006-02-05 13:27:55.751859750 +0100
@@ -3,8 +3,8 @@
 # Override the definition for "ncdump" in "../macros.make".
 NCDUMP          = ../ncdump/ncdump
 
-LIBRARY         = ../libsrc/libnetcdf.a
-LIB_OBJS        = netcdf.o typeSizes.o
+LIBRARYL        = libnetcdf_f90.la
+LIB_OBJS        = netcdf.lo typeSizes.lo
 GARBAGE         = netcdf_test *.mod *.d example.nc example.cdl \
 		  eightByteIntExample.nc
 MANUAL          = netcdf.3f90
@@ -31,7 +31,7 @@
 		nves.m4 nvesput48.m4 nvesget48.m4 \
 		typeSizes.f90 install-sh
 
-all:            $(LIBRARY)
+all:            $(LIBRARYL)
 
 test:           netcdf_test
 	./netcdf_test
@@ -55,23 +55,30 @@
 EightByteIntTest-2:     EightByteIntTest-2.o typeSizes.o
 	$(LINK.F90) $@.o typeSizes.o $(F90LIBS)
 
-typeSizes.o:    typeSizes.f90
-	$(COMPILE.F90) typeSizes.f90
+typeSizes.lo:    typeSizes.f90
+	libtool --mode=compile $(COMPILE.F90) typeSizes.f90
 
 netcdf_expanded.f90:    gen.m4 f90aux.m4 nves.m4 nvea.m4 \
 			nvesput48.m4 nvesget48.m4 nveaput48.m4 nveaget48.m4
 	$(M4) gen.m4 >$@
 
-netcdf.o:       \
-	    netcdf.f90 typeSizes.o netcdf_constants.f90 netcdf_externals.f90 \
+netcdf.lo:       \
+	    netcdf.f90 typeSizes.lo netcdf_constants.f90 netcdf_externals.f90 \
 	    netcdf_overloads.f90 netcdf_visibility.f90 netcdf_file.f90 \
 	    netcdf_dims.f90 netcdf_attributes.f90 netcdf_variables.f90 \
 	    netcdf_text_variables.f90 netcdf_expanded.f90
-	$(COMPILE.F90) netcdf.f90
+	libtool --mode=compile $(COMPILE.F90) netcdf.f90
 
-install:        installed_netcdf_module installed_typesizes_module \
+libnetcdf_f90.la: $(LIB_OBJS)
+	libtool --mode=link $(LINK.F90) $(LIB_OBJS) -rpath $(LIBDIR) ../libsrc/libnetcdf.la
+	
+install:        install-lib installed_netcdf_module installed_typesizes_module \
 		$(MANDIR)/man3f90/$(MANUAL)
 
+install-lib:
+	install -d $(DESTDIR)$(LIBDIR)
+	libtool --mode=install install libnetcdf_f90.la $(DESTDIR)$(LIBDIR)
+
 installed_netcdf_module:
 	@if test -f netcdf.mod; then \
 	    header=netcdf.mod; \
@@ -133,5 +140,5 @@
 .f90.o:
 	$(COMPILE.F90) $<
 
-EightByteIntTest.o:     typeSizes.o
-EightByteIntTest-2.o:   typeSizes.o
+EightByteIntTest.o:     typeSizes.lo
+EightByteIntTest-2.o:   typeSizes.lo
